pennylane/ops/qubit.py

```diff
@@ -530,7 +530,7 @@ class BasisState(Operation):
         wires (Sequence[int] or int): the wire(s) the operation acts on
     """
     num_params = 1
-    num_wires = 0
+    num_wires = Any
```

Comment:
this should be `All` . I also tested it and we get an error if we dont apply this function to all the wires.

Reply:
@AroosaIjaz what's the status of this comment? It seems to have been posted post-merge. Is it still an issue/bug?

Reply:
Yeah, I made an issue about it; issue #286 

I am still testing this function and others on various plugins. So the PR for this issue is still in the making :)